proxy/http2/Http2ConnectionState.cc

```diff
@@ -1160,3 +1162,31 @@ Http2ConnectionState::send_window_update_frame(Http2StreamId id, uint32_t size)
   SCOPED_MUTEX_LOCK(lock, this->ua_session->mutex, this_ethread());
   this->ua_session->handleEvent(HTTP2_SESSION_EVENT_XMIT, &window_update);
 }
+
+// Return min_concurrent_streams_in when current client streams number is larger than max_active_streams_in.
+// Main purpose of this is preventing DDoS Attacks.
+unsigned
+Http2ConnectionState::_adjust_concurrent_stream()
+{
+  int64_t current_client_streams = 0;
+  RecGetRawStatSum(http2_rsb, HTTP2_STAT_CURRENT_CLIENT_STREAM_COUNT, &current_client_streams);
```

Comment:
This is getting the total stream value across all the threads and then you are comparing it to the per thread value below.